### DISCUSSION TITLE
Handling Indexes for Partitions Created in Flyway Migrations

### DISCUSSION THREAD

**Discussion by vbhosle** [pgpartman/pg_partman#621] [General]
We are maintaining flyway migrations along with the code repo. Goal is to use same migrations for unit tests, deploying to all the environments. 
There is a scenario, which we have trouble handling. 
Let's say we create a monthly partition in migration number 1.0, using create_parent with premake as 12.
After few migrations, we decide to add an index. In migration 4.0, we add index on the table. The new partitions will start having the new index. For old partitions we must set it manually, which is fine because we have a context while making a change.
For setting up a new environment, we have automated process to pull the script from repo and apply it. Devs don't have to be consulted.
But with migrations with above scenario, the premake partitions created in 1.0 will not have the index created in 4.0.  Without this context, it easy to have a new environment without critical indexes. How can we ensure that partitions created by 1.0 have the index created in 4.0. We don't want to hardcode dates. 
I think show_partitions should help here. If there is any other way, please let me know.

**Comment by keithf4** [comment]
This is really a specialized requirement for your situation. Not quite sure how partman can help with this since this is more a schema versioning issue. Maybe look at something like sqitch - https://sqitch.org/?
